.github/scripts/cleanup-cluster.sh

```diff
@@ -52,5 +52,14 @@ for ns in $namespaces; do
         break
     fi
     echo "deleting namespaces: $ns"
+
+    # Remove finalizers from the namespace if present
+    kubectl get namespace $ns -o json | jq '.spec.finalizers = []' >temp-namespace.json
```

Comment:
Interestingly, UCP and Applications RP Pods were not healthy, and this caused finalizers to not finalize in some of the resources in some of the namespaces. And then this caused those namespaces to not be deleted.

UCP Pod Down -> Finalizers in some resources -> Namespaces couldn't be deleted -> Clean up step got stuck -> Long Running Workflow failed back to back

Reply:
Do you have logs from the failing pods? or from the Kubernetes `controller-manager` pod?